cnctcli/actions/products/sync.py

```diff
@@ -3,168 +3,44 @@
 # This file is part of the Ingram Micro Cloud Blue Connect connect-cli.
 # Copyright (c) 2019-2020 Ingram Micro. All Rights Reserved.
 
-from collections import namedtuple
 from zipfile import BadZipFile
 
 from click import ClickException
 
 from openpyxl import load_workbook
 from openpyxl.utils.exceptions import InvalidFileException
 
-from tqdm import trange
-
-from cnctcli.actions.products.constants import (
-    BILLING_PERIOD,
-    COMMITMENT,
-    ITEMS_COLS_HEADERS,
-    PRECISIONS,
-)
-from cnctcli.api.products import (
-    create_item,
-    create_unit,
-    delete_item,
-    get_item,
-    get_item_by_mpn,
-    update_item,
+from cnctcli.actions.products.utils import (
+    get_col_limit_by_ws_type,
+    get_ws_type_by_worksheet_name,
+    get_col_headers_by_ws_type,
+    SkipSyncTask,
 )
-from cnct.rql import R
-
-fields = (v.replace(' ', '_').lower() for v in ITEMS_COLS_HEADERS.values())
-
-_RowData = namedtuple('RowData', fields)
 
 
 class ProductSynchronizer:
     def __init__(self, client, silent):
         self._client = client
         self._silent = silent
-        self._units = list(client.ns('settings').units.all())
         self._product_id = None
         self._wb = None
 
-    def open(self, input_file):
+    def open(self, input_file, worksheet):
         self._open_workbook(input_file)
-        if 'Items' not in self._wb.sheetnames:
-            raise ClickException('Invalid input file: File shall contain at least Items sheet.')
-        ws = self._wb[self._wb.sheetnames[0]]
+        if worksheet not in self._wb.sheetnames:
+            raise SkipSyncTask(f'File does not contain {worksheet} to synchronize, skipping')
```

Comment:
Please change the name of this exception. The open should return the sheet called `worksheet` from the workbook to open. open should raise something like SheetNotFound. The caller of open should decide what to do if the sheet does not exist (continue or exit).

Reply:
done